src/Communication/Socket/SocketInterface.php

```diff
@@ -52,4 +52,6 @@ public function isConnected();
      * @return bool
      */
     public function disconnect($reason = 1000);
+
+    public function waitForData(float $maxSeconds): bool;
```

Comment:
adding this is a breaking change. can you make another interface that just has waitForData, implement it, and have callers use an instanceof check please.